src/test/java/hudson/scm/IntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+                        ProcessBuilder pb = new ProcessBuilder("cvs", "-f", "--allow-root=" + repo, "pserver");
+                        final Process server = pb.start();
+                        new Thread("printing errors") {
+                            @Override public void run() {
+                                try {
+                                    copy(server.getErrorStream(), System.err);
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("sending output") {
+                            @Override public void run() {
+                                try {
+                                    copy(server.getInputStream(), /*new TeeOutputStream(*/s.getOutputStream()/*, System.err)*/);
+                                } catch (SocketException x) {
+                                    // Broken pipe? Ignore.
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("accepting input") {
+                            @Override public void run() {
+                                try {
+                                    InputStream is = s.getInputStream();
+                                    OutputStream os = server.getOutputStream();
+                                    copy(is, /*new TeeOutputStream(*/os/*, System.err)*/);
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("wait for exit") {
+                            @Override public void run() {
+                                try {
+                                    int r = server.waitFor();
+                                    if (r != 0) {
+                                        LOGGER.log(Level.INFO, "server exited with status {0}", r);
+                                    }
+                                } catch (Exception x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        };
+                    }
+                } catch (IOException x) {
+                    x.printStackTrace();
+                }
+            }
+        }.start();
+        work = tmp.newFolder();
+        run(command("checkout", "-d", ".", ".").directory(work));
+    }
+
+    @After public void killServer() throws Exception {
+        if (sock != null) {
+            sock.close();
+        }
+        // TODO if necessary, kill all threads & destroy() any server processes still running
```

Comment:
Not quite sure what the request is here, but again I think it falls under the category of “nice things to do but would be wasted effort if we can get Docker fixtures”.